redigomock.go

```diff
@@ -38,6 +38,7 @@ type Conn struct {
 	stats              map[cmdHash]int // Command calls counter
 	statsMut           sync.RWMutex    // Locks the stats so we don't get concurrent map writes
 	Errors             []error         // Storage of all error occured in do functions
+	errorsMut          sync.RWMutex    // Locks Errors to avoid concurrent slice appends
```

Comment:
It's strange to have a mutex on an exported type because someone could have a Go routine reading this attribute without locks. 🤔 Maybe we could follow this pattern:

```go
type Errors struct {
  sync.RWMutex
  errors []error
}

func (e *Errors) Get() []error {
  e.RLock()
  defer e.RUnlock()
  copyErrs := make([]error, len(e.errors))
  copy(copyErrs, e.errors)
  return copyErrs
}

type Conn struct {
  // ...
  Errors Errors // Storage of all error occurred in do functions
}
```

That would be a breaking change though, so we would need to bump the library major version.